src/copy.c

```diff
@@ -892,9 +892,13 @@ copyfrom(CopyChunkState *ccstate, ParseState *pstate, Hypertable *ht, MemoryCont
 
 #if PG16_LT
 	ExecInitRangeTable(estate, pstate->p_rtable);
-#else
+#elif PG18_LT
 	Assert(pstate->p_rteperminfos != NULL);
 	ExecInitRangeTable(estate, pstate->p_rtable, pstate->p_rteperminfos);
+#else
+	/* PG18+ adds unpruned relids to ExecInitRangeTable */
+	Assert(pstate->p_rteperminfos != NULL);
+	ExecInitRangeTable(estate, pstate->p_rtable, pstate->p_rteperminfos, bms_make_singleton(1));
```

Comment:
Let's add some explanation on why it is 1 and not empty for example.

Reply:
i think its arbitrary, i just followed the upstream lead, since this is copy no pruning should be happening

Reply:
i'll add a comment